src/plugins/dashboard/public/application/actions/clone_panel_action.tsx

```diff
@@ -0,0 +1,166 @@
+
+export interface ClonePanelActionContext {
+  embeddable: IEmbeddable;
+}
+
+export class ClonePanelAction implements ActionByType<typeof ACTION_CLONE_PANEL> {
+  public readonly type = ACTION_CLONE_PANEL;
+  public readonly id = ACTION_CLONE_PANEL;
+  public order = 11;
+
+  constructor(private core: CoreStart) {}
+
+  public getDisplayName({ embeddable }: ClonePanelActionContext) {
+    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
+      throw new IncompatibleActionError();
+    }
+    return i18n.translate('dashboard.panel.clonePanel', {
+      defaultMessage: 'Clone panel',
+    });
+  }
+
+  public getIconType({ embeddable }: ClonePanelActionContext) {
+    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
+      throw new IncompatibleActionError();
+    }
+    return 'copy';
+  }
+
+  public async isCompatible({ embeddable }: ClonePanelActionContext) {
+    if (embeddable.getInput().viewMode) {
+      if (embeddable.getInput().viewMode === ViewMode.VIEW) {
+        return false;
+      }
+    }
+
+    return Boolean(
+      embeddable.getRoot() &&
+        embeddable.getRoot().isContainer &&
+        embeddable.getRoot().type === DASHBOARD_CONTAINER_TYPE
+    );
+  }
+
+  public async execute({ embeddable }: ClonePanelActionContext) {
+    if (!embeddable.getRoot() || !embeddable.getRoot().isContainer) {
+      throw new IncompatibleActionError();
+    }
+
+    const dashboard = embeddable.getRoot() as DashboardContainer;
+    const panelToClone = dashboard.getInput().panels[embeddable.id] as DashboardPanelState;
+    if (!panelToClone) {
+      throw new PanelNotFoundError();
+    }
+
+    dashboard.showPlaceholderUntil(
+      this.cloneEmbeddable(panelToClone, embeddable.type),
+      placePanelBeside,
+      {
+        width: panelToClone.gridData.w,
+        height: panelToClone.gridData.h,
+        currentPanels: dashboard.getInput().panels,
+        placeBesideId: panelToClone.explicitInput.id,
+      } as IPanelPlacementBesideArgs
+    );
+  }
+
+  private async getUniqueTitle(rawTitle: string, embeddableType: string): Promise<string> {
+    const clonedTag = i18n.translate('dashboard.panel.title.clonedTag', {
+      defaultMessage: 'copy',
+    });
+    const cloneRegex = new RegExp(`\\(${clonedTag}\\)`, 'g');
+    const cloneNumberRegex = new RegExp(`\\(${clonedTag} [0-9]+\\)`, 'g');
+    const baseTitle = rawTitle
+      .replace(cloneNumberRegex, '')
+      .replace(cloneRegex, '')
+      .trim();
+
+    const similarSavedObjects = await this.core.savedObjects.client.find<SavedObject>({
+      type: embeddableType,
+      perPage: 0,
+      fields: ['title'],
+      searchFields: ['title'],
+      search: `"${baseTitle}"`,
+    });
+    const similarBaseTitlesCount: number = similarSavedObjects.total - 1;
+
+    return similarBaseTitlesCount <= 0
+      ? baseTitle + ` (${clonedTag})`
+      : baseTitle + ` (${clonedTag} ${similarBaseTitlesCount})`;
+  }
+
+  private async cloneEmbeddable(
```

Comment:
Would this be better suited to live on the embeddable itself? In case we have other uses-cases like this? Just thinking out loud, and definitely doesn't need to be made as part of this PR.

Reply:
This method currently seems to work on every different type of embeddable, it would be pretty cool though if embeddables had a way to override it.

Additionally, it could later be moved onto the dashboard embeddable so that it could be re-used if we do need it!